app/services/civil_servant_search.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+class CivilServantSearch
+  class << self
+    def call(term, current_organization)
+      [
+        organization_civil_servant_lookup_group(term, current_organization),
+        all_civil_servant_lookup_group(term),
+        new_civil_servant_lookup_group
+      ]
+    end
+
+    private
+
+    def organization_civil_servant_lookup_group(term, current_organization)
+      {
+        text: I18n.t('organizations.service_agreements.search.modal.dropdown.groups.recent_civil_servants'),
+        children: organization_civil_servant_lookup_results(term, current_organization)
+      }
+    end
+
+    def all_civil_servant_lookup_group(term)
+      {
+        text: I18n.t('organizations.service_agreements.search.modal.dropdown.groups.all_civil_servants'),
+        children: all_civil_servant_lookup_results(term)
+      }
+    end
+
+    def new_civil_servant_lookup_group
+      {
+        text: I18n.t('organizations.service_agreements.search.modal.dropdown.groups.new_civil_servant'),
+        children: [{
+          id: -1,
+          text: I18n.t('organizations.service_agreements.search.modal.dropdown.groups.new_civil_servant')
+        }]
+      }
+    end
+
+    def organization_civil_servant_lookup_results(term, current_organization)
+      select_and_map_civil_servants(filtered_organization_civil_servants(term, current_organization))
+    end
+
+    def all_civil_servant_lookup_results(term)
+      select_and_map_civil_servants(filtered_all_civil_servants(term))
+    end
+
+    def filtered_all_civil_servants(term)
+      if term.present?
+        CivilServant.search(term).distinct
+      else
+        CivilServant.all.distinct.order(:first_name, :last_name)
```

Comment:
Also here, I'd say that the distinct is not needed

Reply:
True but had to add it further down in the select_and_map_civil_servants func